src/experimental/safe_html_transform/tests/test_safe_html.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+import unittest2 as unittest
+from Products.PortalTransforms.data import datastream
```

Comment:
@tisto datastream is part of the Transforms API, so importing it is pretty essential if you're testing that you are compliant with the API.

@prakharjoshi it's not a testing bug per-se. It's the only way to do it given the current structure of ``convert()``, and you have to have at least some tests of this form to make sure you comply with the PortalTransforms API specification. The bigger problem is you're not compliant with this API. If we look at:-

https://github.com/plone/Products.PortalTransforms/blob/master/docs/dev_manual.rst

... it says that your ``convert()`` method should return a datastream object, and it doesn't currently.

Back to @tisto's original comment though. One reason he brings this up is if we want to change that PortalTransforms API in the future, then rewriting all these tests to comply to a new API would be a pain. There are 2 ways around this:

* Split out code in ``convert()`` that purely deals with the transform into a separate function that takes in HTML (and options as a dict) and returns filtered HTML. Then test that your internal function works, and you have interfaced your internal function to the PortalTransforms API correctly separately.
* Pull the common code out of each test in ``test_safe_html.py`` into a helper function, so you only have one bit of code 

I suspect @tisto's approach would tend to the former, mine would tend to the latter. Either is fine.

However, this is drifting away from what this pull request is about. Feel free to copy/paste this discussion onto another pull request where this code is now.